src/_includes/partials/components/navigation.njk

```diff
@@ -7,11 +7,11 @@
                     </li>
                     {% endfor %}
                     {% for item in collections.all %}
-                        {# Find an item with a translation key which matches the current page's translation key but is not the same language. #}
-                        {% if item.data.translationKey == translationKey and item.data.lang !== lang %}
-                        <li lang="{{ item.data.lang }}">
-                            <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
-                        </li>
+{% if item.data.lang !== lang and item.data.translationKey == ('index' if translationKey == 404 else translationKey) %}
+                            <li lang="{{ item.data.lang }}">
+                                <a href="{{ item.url }}">{{ config.languages[item.data.lang].name }}</a>
+                            </li>
+                        {% endif %}
```

Comment:
Some formatting issues from merging the suggestion and a duplicate `{% endif %}`.